docs/designdoc/object-storage-type.md

```diff
@@ -0,0 +1,55 @@
+# Make object storage selectable
+
+## Context
+
+The only object storage interface currently supported by MOCO for storing backups is Amazon S3.
+Amazon S3 interfaces are available for many object storage.
+As a result, MOCO is currently able to support many object storage.
+
+However, there are cases that are not supported.
+
+* [cybozu-go/moco#427](https://github.com/cybozu-go/moco/issues/427)
+
+The aws-sdk-go v2 client used by MOCO cannot use Google Cloud Storage which supports Amazon S3 API compatibility.[^1]
+
+[^1]: https://github.com/aws/aws-sdk-go-v2/issues/1816
+
+If there is motivation to support non-Amazon S3 compatible object storage other than this issue,
+MOCO's current API is not capable of supporting it.
+
+As an example Azure Blob Storage does not support S3 API. (But, proxy exists[^2])
+
+[^2]: https://devblogs.microsoft.com/cse/2016/05/22/access-azure-blob-storage-from-your-apps-using-s3-api/
+
+Therefore, this proposal proposes extensions to MOCO's API to enable support for object storage other than Amazon S3.
+
+## Goals
+
+* No breaking changes
+* Extend API to allow non-Amazon S3 compatible object storage
+
+## Non-goals
+
+* Only proposes API extensions to enable non-S3 compatible object storage
+  * The implementation to support other object storage is carried out in other processes
+
+## ActualDesign
+
+Add `.spec.jobConfig.bucketConfig.backendType` field to BackupPolicy.
+
+```yaml
+apiVersion: moco.cybozu.com/v1beta2
+kind: BackupPolicy
+metadata:
+  namespace: backup
+  name: daily
+spec:
+  jobConfig:
+...
+    bucketConfig:
+      backendType: s3 # added
```

Comment:
Will `s3` be the default?

Reply:
Yes.
I think it is appropriate to default to the S3 API as it has become the de facto standard interface for almost all storage providers.

I'm considering setting the string `s3` if empty in the mutating webhook.

Reply:
Good. Note also that existing BackupPolicy might have empty `backendType`,
so MOCO should treat empty as S3.